pkg/autoscaling/estimator/percentile.go

```diff
@@ -66,28 +65,71 @@ func (e *PercentileResourceEstimator) GetResourceEstimation(evpa *autoscalingapi
 				Namespace:     evpa.Namespace,
 				WorkloadName:  evpa.Spec.TargetRef.Name,
 				ContainerName: containerName,
-				Selector:      labels.Everything(),
+				Selector:      selector,
 			},
 		},
 	}
-
 	memConfig := getMemConfig(config)
-	tsList, err = utils.QueryPredictedValues(e.Predictor, caller, memConfig, memoryMetricNamer)
+
+	var errs []error
+	// first register cpu & memory, or the memory will be not registered before the cpu prediction succeed
```

Comment:
Would you please make this commend more clearer.

Reply:
> Would you please make this commend more clearer.

ok. originally, we call `utils.QueryPredictedValues` it will do register & predict. then it failed for cpu, then it will be always return directly, but the memory is not registered forever before the cpu is succeed. 

so, for each predicting, we should first to register what you need to predict, then do predicting. because evpa maybe has multiple metrics for predicting, so we first register all of them, then get predicted value by reconciling

I will make it clear more